### ISSUE TITLE
Could not find 'appInsights.updateCfg' (Blazor Web Assembly stand-alone)

### ISSUE THREAD

**Issue by robertwood62** [IvanJosipovic/BlazorApplicationInsights#244]
I just upgraded to 3.0.0 to take advantage of the delayed loading and initializing of the Connection String and Key.  But I'm getting an error in the browser.

**Error Message in Browser:**
Microsoft.AspNetCore.Components.WebAssembly.Rendering.WebAssemblyRenderer[100]
      Unhandled exception rendering component: Could not find 'appInsights.updateCfg' ('updateCfg' was undefined).
      Error: Could not find 'appInsights.updateCfg' ('updateCfg' was undefined).
          at https://localhost:8080/_framework/blazor.webassembly.js:1:328
          at Array.forEach (<anonymous>)
          at a.findFunction (https://localhost:8080/_framework/blazor.webassembly.js:1:296)
          at _ (https://localhost:8080/_framework/blazor.webassembly.js:1:2442)
          at https://localhost:8080/_framework/blazor.webassembly.js:1:3330
          at new Promise (<anonymous>)
          at Object.beginInvokeJSFromDotNet (https://localhost:8080/_framework/blazor.webassembly.js:1:3311)
          at Object.Gt [as invokeJSFromDotNet] (https://localhost:8080/_framework/blazor.webassembly.js:1:62569)
          at Object.Ii (https://localhost:8080/_framework/dotnet.7.0.14.b1t3rmny8d.js:5:71974)
          at _mono_wasm_invoke_js_blazor (https://localhost:8080/_framework/dotnet.7.0.14.b1t3rmny8d.js:14:103886)
Microsoft.JSInterop.JSException: Could not find 'appInsights.updateCfg' ('updateCfg' was undefined).
Error: Could not find 'appInsights.updateCfg' ('updateCfg' was undefined).
    at https://localhost:8080/_framework/blazor.webassembly.js:1:328
    at Array.forEach (<anonymous>)
    at a.findFunction (https://localhost:8080/_framework/blazor.webassembly.js:1:296)
    at _ (https://localhost:8080/_framework/blazor.webassembly.js:1:2442)
    at https://localhost:8080/_framework/blazor.webassembly.js:1:3330
    at new Promise (<anonymous>)
    at Object.beginInvokeJSFromDotNet (https://localhost:8080/_framework/blazor.webassembly.js:1:3311)
    at Object.Gt [as invokeJSFromDotNet] (https://localhost:8080/_framework/blazor.webassembly.js:1:62569)
    at Object.Ii (https://localhost:8080/_framework/dotnet.7.0.14.b1t3rmny8d.js:5:71974)
    at _mono_wasm_invoke_js_blazor (https://localhost:8080/_framework/dotnet.7.0.14.b1t3rmny8d.js:14:103886)
   at Microsoft.JSInterop.JSRuntime.<InvokeAsync>d__16`1[[Microsoft.JSInterop.Infrastructure.IJSVoidResult, Microsoft.JSInterop, Version=7.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60]].MoveNext()
   at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
   at BlazorApplicationInsights.ApplicationInsights.UpdateCfg(Config newConfig, Nullable`1 mergeExisting)
   at BlazorApplicationInsights.ApplicationInsightsInit.OnAfterRenderAsync(Boolean firstRender)
   at Microsoft.AspNetCore.Components.RenderTree.Renderer.GetErrorHandledTask(Task taskToHandle, ComponentState owningComponentState)

### Program.cs
builder.Services.AddBlazorApplicationInsights(config =>
{
    config.ConnectionString = environment.AppInsightsConnectionString;
    config.InstrumentationKey = environment.AppInsightsInstrumentationId;
},
async applicationInsights =>
{
    await applicationInsights.AddTelemetryInitializer(new TelemetryItem()
    {
        Tags = new Dictionary<string, object?>()
        {
            { "ai.cloud.role", "SPA" },
            { "ai.cloud.roleInstance", "SLP" },
        }
    });
});

### Index.html
        <base href="/" />
        <script type="text/javascript">
            !(function (cfg) { function e() { cfg.onInit && cfg.onInit(i) } var S, u, D, t, n, i, C = window, x = document, w = C.location, I = "script", b = "ingestionendpoint", E = "disableExceptionTracking", A = "ai.device."; "instrumentationKey"[S = "toLowerCase"](), u = "crossOrigin", D = "POST", t = "appInsightsSDK", n = cfg.name || "appInsights", (cfg.name || C[t]) && (C[t] = n), i = C[n] || function (l) { var d = !1, g = !1, f = { initialize: !0, queue: [], sv: "7", version: 2, config: l }; function m(e, t) { var n = {}, i = "Browser"; function a(e) { e = "" + e; return 1 === e.length ? "0" + e : e } return n[A + "id"] = i[S](), n[A + "type"] = i, n["ai.operation.name"] = w && w.pathname || "_unknown_", n["ai.internal.sdkVersion"] = "javascript:snippet_" + (f.sv || f.version), { time: (i = new Date).getUTCFullYear() + "-" + a(1 + i.getUTCMonth()) + "-" + a(i.getUTCDate()) + "T" + a(i.getUTCHours()) + ":" + a(i.getUTCMinutes()) + ":" + a(i.getUTCSeconds()) + "." + (i.getUTCMilliseconds() / 1e3).toFixed(3).slice(2, 5) + "Z", iKey: e, name: "Microsoft.ApplicationInsights." + e.replace(/-/g, "") + "." + t, sampleRate: 100, tags: n, data: { baseData: { ver: 2 } }, ver: 4, seq: "1", aiDataContract: undefined } } var h = -1, v = 0, y = ["js.monitor.azure.com", "js.cdn.applicationinsights.io", "js.cdn.monitor.azure.com", "js0.cdn.applicationinsights.io", "js0.cdn.monitor.azure.com", "js2.cdn.applicationinsights.io", "js2.cdn.monitor.azure.com", "az416426.vo.msecnd.net"], k = l.url || cfg.src; if (k) { if ((n = navigator) && (~(n = (n.userAgent || "").toLowerCase()).indexOf("msie") || ~n.indexOf("trident/")) && ~k.indexOf("ai.3") && (k = k.replace(/(\/)(ai\.3\.)([^\d]*)$/, function (e, t, n) { return t + "ai.2" + n })), !1 !== cfg.cr) for (var e = 0; e < y.length; e++)if (0 < k.indexOf(y[e])) { h = e; break } var i = function (e) { var a, t, n, i, o, r, s, c, p, u; f.queue = [], g || (0 <= h && v + 1 < y.length ? (a = (h + v + 1) % y.length, T(k.replace(/^(.*\/\/)([\w\.]*)(\/.*)$/, function (e, t, n, i) { return t + y[a] + i })), v += 1) : (d = g = !0, o = k, c = (p = function () { var e, t = {}, n = l.connectionString; if (n) for (var i = n.split(";"), a = 0; a < i.length; a++) { var o = i[a].split("="); 2 === o.length && (t[o[0][S]()] = o[1]) } return t[b] || (e = (n = t.endpointsuffix) ? t.location : null, t[b] = "https://" + (e ? e + "." : "") + "dc." + (n || "services.visualstudio.com")), t }()).instrumentationkey || l.instrumentationKey || "", p = (p = p[b]) ? p + "/v2/track" : l.endpointUrl, (u = []).push((t = "SDK LOAD Failure: Failed to load Application Insights SDK script (See stack for details)", n = o, r = p, (s = (i = m(c, "Exception")).data).baseType = "ExceptionData", s.baseData.exceptions = [{ typeName: "SDKLoadFailed", message: t.replace(/\./g, "-"), hasFullStack: !1, stack: t + "\nSnippet failed to load [" + n + "] -- Telemetry is disabled\nHelp Link: https://go.microsoft.com/fwlink/?linkid=2128109\nHost: " + (w && w.pathname || "_unknown_") + "\nEndpoint: " + r, parsedStack: [] }], i)), u.push((s = o, t = p, (r = (n = m(c, "Message")).data).baseType = "MessageData", (i = r.baseData).message = 'AI (Internal): 99 message:"' + ("SDK LOAD Failure: Failed to load Application Insights SDK script (See stack for details) (" + s + ")").replace(/\"/g, "") + '"', i.properties = { endpoint: t }, n)), o = u, c = p, JSON && ((r = C.fetch) && !cfg.useXhr ? r(c, { method: D, body: JSON.stringify(o), mode: "cors" }) : XMLHttpRequest && ((s = new XMLHttpRequest).open(D, c), s.setRequestHeader("Content-type", "application/json"), s.send(JSON.stringify(o)))))) }, a = function (e, t) { g || setTimeout(function () { !t && f.core || i() }, 500), d = !1 }, T = function (e) { var n = x.createElement(I), e = (n.src = e, cfg[u]); return !e && "" !== e || "undefined" == n[u] || (n[u] = e), n.onload = a, n.onerror = i, n.onreadystatechange = function (e, t) { "loaded" !== n.readyState && "complete" !== n.readyState || a(0, t) }, cfg.ld && cfg.ld < 0 ? x.getElementsByTagName("head")[0].appendChild(n) : setTimeout(function () { x.getElementsByTagName(I)[0].parentNode.appendChild(n) }, cfg.ld || 0), n }; T(k) } try { f.cookie = x.cookie } catch (p) { } function t(e) { for (; e.length;)!function (t) { f[t] = function () { var e = arguments; d || f.queue.push(function () { f[t].apply(f, e) }) } }(e.pop()) } var r, s, n = "track", o = "TrackPage", c = "TrackEvent", n = (t([n + "Event", n + "PageView", n + "Exception", n + "Trace", n + "DependencyData", n + "Metric", n + "PageViewPerformance", "start" + o, "stop" + o, "start" + c, "stop" + c, "addTelemetryInitializer", "setAuthenticatedUserContext", "clearAuthenticatedUserContext", "flush"]), f.SeverityLevel = { Verbose: 0, Information: 1, Warning: 2, Error: 3, Critical: 4 }, (l.extensionConfig || {}).ApplicationInsightsAnalytics || {}); return !0 !== l[E] && !0 !== n[E] && (t(["_" + (r = "onerror")]), s = C[r], C[r] = function (e, t, n, i, a) { var o = s && s(e, t, n, i, a); return !0 !== o && f["_" + r]({ message: e, url: t, lineNumber: n, columnNumber: i, error: a, evt: C.event }), o }, l.autoExceptionInstrumented = !0), f }(cfg.cfg), (C[n] = i).queue && 0 === i.queue.length ? (i.queue.push(e), i.trackPageView({})) : e(); })({
                src: "https://js.monitor.azure.com/scripts/b/ai.3.gbl.min.js",
                ld: -1,
                crossOrigin: "anonymous",
                cfg: {
                    instrumentationKey: "00000000-0000-0000-0000-000000000000",
                    disableTelemetry: true
                }
            });
        </script>    

### App.razor
@inject NavigationManager Navigation
@inject IStringLocalizer<Strings> Localizer

<ApplicationInsightsInit IsWasmStandalone="true" />
<CascadingAuthenticationState>
    <Router AppAssembly="@typeof(App).Assembly">
        <Found Context="routeData">
            <AuthorizeRouteView RouteData="@routeData" DefaultLayout="@typeof(MainLayout)">
                <NotAuthorized>

**Comment by robertwood62** [comment]
For some reason the App.razor was cut off...  Here is the relevant fragment.
`
@inject NavigationManager Navigation
@inject IStringLocalizer<Strings> Localizer

<ApplicationInsightsInit IsWasmStandalone="true" />
<CascadingAuthenticationState>
    <Router AppAssembly="@typeof(App).Assembly">
        <Found Context="routeData">
            <AuthorizeRouteView RouteData="@routeData" DefaultLayout="@typeof(MainLayout)">
                <NotAuthorized>`

**Comment by robertwood62** [comment]
Another small clue in my browser messages...

Init.js:24  Failed to initialize AppInsights JS SDK for instance appInsights - V1 API compatibility is no longer supported
nh @ Init.js:24

My key and connection string are as follows also:

## Connection String
InstrumentationKey=fea1e530-927f-4972-b3e5-63c491614e81;IngestionEndpoint=https://canadacentral-1.in.applicationinsights.azure.com/;LiveEndpoint=https://canadacentral.livediagnostics.monitor.azure.com/

## Stand alone key (same as above)
fea1e530-927f-4972-b3e5-63c491614e81

**Comment by IvanJosipovic** [comment]
Could you please provide a sample project exhibiting this issue?

**Comment by robertwood62** [comment]
I figured it out.  Deep inside one of my scripts I had a 'appInsights' property defined which was conflicting with the initialization of Application Insights (so dumb).  I think one one point this code was going to be used to help set the configuration dynamically before you added this capability to the current version.  This bit of legacy code I failed to remove was the issue.  Thanks so much for your quick response and building this very useful Blazor component.
